types/hapi__hapi/index.d.ts

```diff
@@ -3964,6 +3964,11 @@ export class Server {
     validator(joi: Root): void;
 }
 
+/**
+ * Factory function to create a new server object (introduced in v17).
+ */
+export function server(opts?: ServerOptions): Server;
```

Comment:
Funfact: Both `server` and `Server` are the same, they are *all* factory functions.
Calling `new Server()` only works since returning an `object` type value from a constructor overrides the return value.
See https://github.com/hapijs/hapi/blob/master/lib/index.js#L9